### ISSUE TITLE
Discord RPC not working on steam deck

### ISSUE THREAD

**Issue by zqpvr** [flathub/io.github.vinegarhq.Vinegar#57]
I am using the flatpak version of Discord and Vinegar.

**Comment by zzahkaboom24** [comment]
To the devs seeing this. Upon testing, xdg-run/app/com.discordapp.Discord/discord-ipc-0 is wrong. It needs access to te whole directory, not only the single file. Tho I will also see if it still works after a restart

To @zqpvr, download Flatseal, if thats easier for you, click on Vinegar, scroll down until you see the following 2 texts and set them to the same value that I have set them

![image](https://github.com/flathub/io.github.vinegarhq.Vinegar/assets/42496614/5b65fb51-606d-44ce-815f-86d6b5c7204f)

**Comment by zzahkaboom24** [comment]
> Hmm... Issue seems to still persist. This also happens with steam games running under proton-ge and regular proton.

Yea, the command only works only until you reboot. And you have to make sure to set the paths in flatseal the way I set them. Also make sure discord_rpc is set to true

![image](https://github.com/flathub/io.github.vinegarhq.Vinegar/assets/42496614/5e0d9795-f78a-4634-9216-58eb78c49ad2)

**Comment by zzahkaboom24** [comment]
I am a bit at a loss. The script from Reddit works and upon reboot, the file gets created. Everything works the way I set it up, but only if Discord starts first upon reboot. If I start Roblox using Vinegar first, and then Discord, it may lock up and Vinegar may tell me "bwrap: Can't make symlink at /run/user/1000/discord-ipc-0: File exists". So making sure Discord starts first before starting Roblox may be something to keep in mind

**Comment by Thelolguy1** [comment]
okay, I assume #61 fixes this, so I will close this for now. If the issue returns, you are free to reopen.

**Comment by zzahkaboom24** [comment]
@zqpvr I assume wiht the "doesn't seem to work" you mean you tried launching it from the normal application menu and it didnt launch. If thats true, go to `/home/<Your User>/.local/share/applications/`, edit the `io.github.vinegarhq.Vinegar.app.desktop` file with any text editor you prefer and change `--branch=stable` to `--branch=test` in the Exec line. Then it should work. Or to avoid all this work, simply run `flatpak run io.github.vinegarhq.Vinegar player` (provided, you already ran `flatpak run io.github.vinegarhq.Vinegar edit` and added  

[player]  
discord_rpc = true  

to your config)

**Comment by zzahkaboom24** [comment]
But when testing my test build, the file does not get created. So I assume that test build that the flathubbot created did not take my change in for some reason?

Cloning this repository and building it using `make` should work tho. The changes are merged into this after all

**Comment by zzahkaboom24** [comment]
@zqpvr No, the config is empty by default. Regardless. RPC still wont work. I realized that Thelolguy1 put the command needed for the discord files in the bundle-setup module, and I think this only works on build, not on launch. I will move up the command to be executed with the vinegar command instead and will test if that works. No wonder why the test build did not work xd. Here was Thelolguy1's code

modules:
  - name: bundle-setup
    buildsystem: simple
    build-commands:
      - mkdir -p ${FLATPAK_DEST}/{,lib/debug/}lib/i386-linux-gnu/GL
      - mkdir -p ${FLATPAK_DEST}/dlls
      - mkdir /app/dlls/lib -p
      - mkdir /app/dlls/lib32 -p
      - mkdir -p /app/lib32
      - mkdir -p /app/lib/i386-linux-gnu
      - install -Dm644 ld.so.conf -t /app/etc/
      - |
        for i in {0..9}; do
          test -S $XDG_RUNTIME_DIR/discord-ipc-$i || ln -sf {app/com.discordapp.Discord,$XDG_RUNTIME_DIR}/discord-ipc-$i;
        done  
    sources:
      - type: file
        path: ld.so.conf

**Comment by zzahkaboom24** [comment]
@Thelolguy1 small notification so you know about this too, the issue should be reopened, I will open a PR after I concluded that it creates said files

**Comment by zzahkaboom24** [comment]
@zqpvr You may try the PR I pushed this time after the test is available, and you were right, discordrpc is true by default. I had it wrong in my mind

**Comment by zzahkaboom24** [comment]
> Still doesn't work, do I need to do the flatseal thing?

it should be done automatically under normal circumstances, since thats what the yml file already has

**Comment by zzahkaboom24** [comment]
I am unsure why it doesnt work for you apparently, because upon trying it works with no issues on both my laptop and my pc

**Comment by zzahkaboom24** [comment]
Yea no, I think its either a Steam Deck issue, or I dont see why its not working. Assuming you properly deleted the data just in case with `flatpak uninsall --delete-data vinegar` (or io.github.vinegarhq.Vinegar) and did not go into gaming mode, it should work

**Comment by zzahkaboom24** [comment]
@zqpvr I assume you launched via the desktop files. Try launching from the terminal instead, thanks letting me know

**Comment by zzahkaboom24** [comment]
@zqpvr You may try out the same PR, but the newest test install with the number 61426 at #64. I should have basically every case covered automatically

**Comment by zzahkaboom24** [comment]
nice, finally :D
